dessia_common/workflow/core.py

```diff
@@ -1941,6 +1943,138 @@ def _method_jsonschemas(self):
         warnings.warn("method_jsonschema method is deprecated. Use method_schema instead", DeprecationWarning)
         return self.method_schemas
 
+    @staticmethod
+    def instantiate_objects(objects):
+        """ Instantiates objects from a list of objects. """
+        class_name = f"{objects[0].__class__.__name__}"
+        if len(objects) > 1:
+            instances = '['
+            for _ in objects:
+                inst = f"{class_name}('Put here your object')"
+                instances += inst + ", "
+            instances += ']'
+            return instances
+
+        return f"{class_name}('Put here your object')"
+
+    @staticmethod
+    def get_files(type_files):
+        """ Generates file instances from a list of file types. """
+        file_class = type_files[0].__class__.__name__
+        if len(type_files) > 1:
+            instances = "["
+            for _ in range(len(type_files)):
+                instances += f"{file_class}.from_file('Insert your path here')" + ", "
+            instances += ']'
+            return instances
+        return f"{file_class}.from_file('Insert your path here')"
+
+    @staticmethod
+    def is_object_sequence(seq):
+        """ Checks if a sequence contains serializable objects. """
+        return any(isinstance(element, SerializableObject) for element in seq)
+
+    @staticmethod
+    def is_file_sequence(sequence):
+        """ Checks if a sequence contains binary files. """
+        return any(isinstance(element, BinaryFile) for element in sequence)
+
+    def get_sequence(self, sequence):
+        """ Processes a sequence based on its content. """
+        if self.is_object_sequence(sequence):
+            value = self.instantiate_objects(sequence)
+        elif self.is_file_sequence(sequence):
+            value = self.get_files(sequence)
+        else:
+            value = repr(sequence)
+        return value
+
+    def _compute_input_values(self):
+        """ Computes the input values for blocks and non-block variables. """
+        values_nonblock_variables = []
+        values_blocks = []
```

Comment:
Consider : 

```py
nbv_values = [self.input_values[i] for i, v in enumerate(self.workflow.inputs) if v in self.workflow.nonblock_variables]
block_values = [self.input_values[i] for i, v in enumerate(self.workflow.inputs) if v not in self.workflow.nonblock_variables
        return values_nonblock_variables, values_blocks

```

OR

```py
nbv_values = []
block_values = []
for i, variable in enumerate(self.workflow.inputs):
    sequence = nbv_values if variable in self.workflow.non_block_variables else block_values
    sequence.push(self.input_values[i]
return nbv_values, block_values
```

Note :
- nbv is considered ok as part of workflow's jargon
- abbreviating variable to single letter names is considered ok in list comprehension

Reply:
done !